src/metalperformanceshaders.cs

```diff
@@ -1135,5 +1541,1851 @@ interface MPSMatrixMultiplication {
 		void EncodeToCommandBuffer (IMTLCommandBuffer commandBuffer, MPSMatrix leftMatrix, MPSMatrix rightMatrix, MPSMatrix resultMatrix);
 
 		// [Export ("initWithDevice:")] marked as NS_UNAVAILABLE - Use the above initialization method instead.
+
+		// inlining ctor from base class
+		[TV (11,0), iOS (11,0)]
+		[Export ("initWithCoder:device:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (NSCoder aDecoder, IMTLDevice device);
+
+		[TV (11,0), iOS (11,0)]
+		[Export ("batchStart")]
+		nuint BatchStart { get; set; }
+
+		[TV (11,0), iOS (11,0)]
+		[Export ("batchSize")]
+		nuint BatchSize { get; set; }
+	}
+
+	[TV (11,0), Mac (10, 13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface MPSState {
+		[Export ("readCount")]
+		nuint ReadCount { get; set; }
+
+		[Export ("isTemporary")]
+		bool IsTemporary { get; }
+
+		[NullAllowed, Export ("label")]
+		string Label { get; set; }
+	}
+
+	[TV (11,0), Mac (10, 13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof(MPSMatrix))]
+	[DisableDefaultCtor]
+	interface MPSTemporaryMatrix {
+		[Static]
+		[Export ("temporaryMatrixWithCommandBuffer:matrixDescriptor:")]
+		MPSTemporaryMatrix Create (IMTLCommandBuffer commandBuffer, MPSMatrixDescriptor matrixDescriptor);
+
+		[Static]
+		[Export ("prefetchStorageWithCommandBuffer:matrixDescriptorList:")]
+		void PrefetchStorage (IMTLCommandBuffer commandBuffer, MPSMatrixDescriptor[] descriptorList);
+
+		[Export ("readCount")]
+		nuint ReadCount { get; set; }
+	}
+
+	[TV (11,0), Mac (10, 13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface MPSVector {
+		[Export ("device", ArgumentSemantic.Retain)]
+		IMTLDevice Device { get; }
+
+		[Export ("length")]
+		nuint Length { get; }
+
+		[Export ("vectors")]
+		nuint Vectors { get; }
+
+		[Export ("dataType")]
+		MPSDataType DataType { get; }
+
+		[Export ("vectorBytes")]
+		nuint VectorBytes { get; }
+
+		[Export ("data")]
+		IMTLBuffer Data { get; }
+
+		[Export ("initWithBuffer:descriptor:")]
+		IntPtr Constructor (IMTLBuffer buffer, MPSVectorDescriptor descriptor);
+	}
+
+	[TV (11,0), Mac (10, 13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof (NSObject))]
```

Comment:
likely missing a `DisableDefaultCtor` considering there's many readonly properties and there are static methods to create instances